io/src/main/scala/fs2/io/file/package.scala

```diff
@@ -0,0 +1,249 @@
+package fs2.io
+
+import java.nio.ByteBuffer
+import java.nio.channels.{AsynchronousFileChannel, CompletionHandler, FileChannel, FileLock}
+import java.nio.file.{OpenOption, Path, StandardOpenOption}
+
+import fs2.Stream.Handle
+import fs2._
+import fs2.util.Monad
+
+trait FileHandle[F[_]] {
+  type Lock
+
+  // Close the FileHandle
```

Comment:
Will do, I'll ping when that's pushed. Other ideas welcome, but I think it's fairly usable now.